### PR TITLE
(FM-2564) Update generated types part 3

### PR DESCRIPTION
**Pull request by jpogran**
Updates all DSC Resources from PowerShell/DSCResources github repo

Let's add a checklist of all the stuff that needs to be done here:

#### Upstream repo PRs

Based on https://github.com/Iristyle/puppetlabs-dsc/commit/99385869a0a093a2bb70ea65e4bf449c9c9ee650 and comments at https://github.com/puppetlabs/puppetlabs-dsc/pull/22#issuecomment-126720376

- [x] xAzure - https://github.com/PowerShell/xAzure/pull/1 - merged in https://github.com/PowerShell/xAzure/commit/1e6f281ef7fe759ef415fdc3da4b8be582f36b2d - SHA 1e6f281ef7fe759ef415fdc3da4b8be582f36b2d
- [x] xDnsServer - https://github.com/PowerShell/xDnsServer/pull/4 - merged in https://github.com/PowerShell/xDnsServer/commit/71a5ccbce5602cbed0cccb585e33e6e739b209ed - SHA 71a5ccbce5602cbed0cccb585e33e6e739b209ed
- [x] xJea - https://github.com/PowerShell/xJea/pull/4 - merged in https://github.com/PowerShell/xJea/commit/92edf3d45449dcb4b84650d73c79ae7da93b368a - SHA 92edf3d45449dcb4b84650d73c79ae7da93b368a
- [x] xPSDesiredStateConfiguration - https://github.com/PowerShell/xPSDesiredStateConfiguration/pull/37 - merged in https://github.com/PowerShell/xPSDesiredStateConfiguration/commit/cd501a6ab15f53a6e3c9aabf7d1289817e5ca1c9 - SHA cd501a6ab15f53a6e3c9aabf7d1289817e5ca1c9
- [x] xRemoteDesktopSessionHost - https://github.com/PowerShell/xRemoteDesktopSessionHost/pull/1 - merged in https://github.com/PowerShell/xRemoteDesktopSessionHost/commit/40cb00a05de26a879e1123308dff3eda79e1a9ba - SHA 40cb00a05de26a879e1123308dff3eda79e1a9ba
- [x] xSystemSecurity - https://github.com/PowerShell/xSystemSecurity/pull/3 - merged in https://github.com/PowerShell/xSystemSecurity/commit/3d5dbc30db675b87e6d9bb0679e9fab30d46d4fc - SHA 3d5dbc30db675b87e6d9bb0679e9fab30d46d4fc

James also found

- [x] xExchange - https://github.com/PowerShell/xExchange/pull/22 - merged in https://github.com/PowerShell/xExchange/commit/ac40f700531d8dc1d7a23ad9066276e904a4bbbd - SHA ac40f700531d8dc1d7a23ad9066276e904a4bbbd
- [x] xHyper-V - https://github.com/PowerShell/xHyper-V/pull/19 - merged in https://github.com/PowerShell/xHyper-V/commit/653647912b6f07151c1189af24a84301bfc3fe46 - SHA 653647912b6f07151c1189af24a84301bfc3fe46
- [x] xStorage - https://github.com/PowerShell/xStorage/pull/6 - merged in https://github.com/PowerShell/xStorage/commit/60c3bb4d911402bbc19d13dbfbc7ce54c613964b - SHA 60c3bb4d911402bbc19d13dbfbc7ce54c613964b
- [ X] xWinEventLog - https://github.com/PowerShell/xWinEventLog/pull/8 - merged in https://github.com/PowerShell/xWinEventLog/commit/6f0998600e04371264a2d56c471fb59526996631 -SHA 6f0998600e04371264a2d56c471fb59526996631

Update to submodule references
- [x] DscResources - https://github.com/PowerShell/DscResources/pull/50 - merged in https://github.com/PowerShell/DscResources/commit/6840626a531e28393cf1fcb13d7603caf46de217 - SHA 6840626a531e28393cf1fcb13d7603caf46de217
- [X] DscResources - https://github.com/PowerShell/DscResources/pull/52 - merged in https://github.com/PowerShell/DscResources/commit/dc5876d4cad9dce452106a79e2ab6e4f72c5b96b - SHA dc5876d4cad9dce452106a79e2ab6e4f72c5b96b 

Supersedes #51.

### CODE REVIEW

**Comment by jpogran** [comment]
My take-away was first pass to get the rake tasks the way we wanted them and have it pull all of the repos. Then next step is setup a system to choose specific ones when we implement MODULES-2244

File: build/dsc.rake [hunk 1]
```diff
@@ -93,6 +105,7 @@ eod
       dsc_resources_path = args[:dsc_resources_path] || default_dsc_resources_path
       puts "Cleaning #{item_name}"
       sh "librarian-repo clean --verbose --path #{dsc_resources_path}"
```
Comment (ferventcoder) [inline build/dsc.rake]: Does this need to be removed?

**Comment by Iristyle** [comment]
Just an FYI, the previous workflow doesn't really work anymore:

```
bundle exec rake dsc:clean
bundle exec rake dsc:build
```

Breaking out to individual tasks is great, but I think we want to make sure that sequence of tasks still works OK.

**Comment by ferventcoder** [comment]
> Breaking out to individual tasks is great, but I think we want to make sure that sequence of tasks still works OK.

@Iristyle I think we were going for 

~~~
bundle exec rake dsc::clean
bundle exec rake dsc::import
bundle exec rake dsc::build
~~~

but what I gather is you want the import to happen in build if the import has not yet happened.

**Comment by ferventcoder** [comment]
@jpogran I believe Ethan stated those WIP commits he put in here could be completely removed.

**Comment by jpogran** [comment]
Commits removed and rebased

**Comment by ferventcoder** [comment]
This is good. I was initially confused because it doesn't actually copy the updated resources over to the puppet_x folder. So I think the followup should take care of that and all of the new type updates.

**Pull request merged**